ping/README.md

```diff
@@ -0,0 +1,161 @@
+# A simple libp2p ping Testground test plan
+
+This is a simple Testground test plan that was created for instructional purposes.
+It is commented in a step-by-step didactic way to teach folks how to use Testground.
+
+**Contents**
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [About this test plan](#about-this-test-plan)
+- [Installing the test plan](#installing-the-test-plan)
+- [Running a single-group run with defaults](#running-a-single-group-run-with-defaults)
+- [Running a single-group run with parameters](#running-a-single-group-run-with-parameters)
+- [Running a composition](#running-a-composition)
+- [Running with local:exec and cluster:k8s runners](#running-with-localexec-and-clusterk8s-runners)
+- [License](#license)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## About this test plan
+
+This plan contains a single test case called `ping` (same as the plan).
+
+This test case supports three test parameters:
+
+* `secure_channel`: the secure channel to use. One of: `noise`, `secio`, `tls` (default: `noise`).
+* `max_latency_ms`: the maximum latency to set in traffic shaping, in milliseconds (default: `1000`).
+* `iterations`: the number of ping iterations to run (default: `5`).
+
+This test case:
+
+1. Instantiates a libp2p Host, using the instructed secure channel.
+2. Connects to all other Hosts participating in the test plan.
+3. Does an initial round of pings via the libp2p PING protocol, outputting the
+   observed latencies as log messages and result data points. This round is not
+   counted against the `iterations` param.
+4. It then requests a network configuration change to set a random latency
+   (up to `max_latency_ms`) on our link.
+5. Once all instances have had their latencies applied, we perform a round of 
+   PINGs and record the observed latencies, just like in point 3.
+6. We do 4-6 as many times as `iterations`.
+
+## Installing the test plan
+
+First, install Testground:
+
+```shell script
+$ git clone https://github.com/testground/testground.git
+
+$ cd testground
+
+$ make install       # builds testground and the Docker image, used by the local:docker runner.
+
+$ testground daemon  # will start the daemon listening on localhost:8042 by default.
+# => this will run indefinitely; open a different console to continue.
+```
+
+Now, import this test plan into your `TESTGROUND_HOME` dir (by default `$HOME/testground`):
+
+```shell script
+# clone this repo in your directory of choice; let's call it `$REPO_DIR`
```

Comment:
Maybe just use `go get` and assume this will be in `GOPATH` to simplify and give a ready-to-use command to the user?

Reply:
Hum, the go get command has way too many side effects depending on what your `PWD` is at that moment. I'd refrain from doing that.